### PR TITLE
Add success action status to PresignedPostPolicy

### LINKED ISSUE #782: PostPolicy not implements  SetSuccessStatusAction?
PostPolicy not implements  SetSuccessStatusAction?

### PR DESCRIPTION
**Pull request by sinhaashish**
Parameter added to set `success-action-status` for making PresignedPostPolicy request.

The generated curl command : 
``` 
curl -X POST  -F bucket=testbucket -F x-amz-date=20190804T172411Z -F success_action_status=201 -F x-amz-signature=c9358c3dc8c5e23b451c1a248f6523e44daeb1a01bd63999b3f389e6a8345fc4 -F key=my-objectname -F x-amz-algorithm=AWS4-HMAC-SHA256 -F Content-Type=image/png -F x-amz-credential=Q3AM3UQ867SPQQA43P2F/20190804/us-east-1/s3/aws4_request -F policy=eyJleHBpcmF0aW9uIjoiMjAxOS0wOC0xMVQxNzoyNDoxMC4xNzNaIiwiY29uZGl0aW9ucyI6W1siZXEiLCIkYnVja2V0IiwidGVzdGJ1Y2tldCJdLFsiZXEiLCIka2V5IiwibXktb2JqZWN0bmFtZSJdLFsiZXEiLCIkQ29udGVudC1UeXBlIiwiaW1hZ2UvcG5nIl0sWyJlcSIsInN1Y2Nlc3NfYWN0aW9uX3N0YXR1cyIsIjIwMCJdLFsiZXEiLCIkeC1hbXotYWxnb3JpdGhtIiwiQVdTNC1ITUFDLVNIQTI1NiJdLFsiZXEiLCIkeC1hbXotY3JlZGVudGlhbCIsIlEzQU0zVVE4NjdTUFFRQTQzUDJGLzIwMTkwODA0L3VzLWVhc3QtMS9zMy9hd3M0X3JlcXVlc3QiXSxbImVxIiwiJHgtYW16LWRhdGUiLCIyMDE5MDgwNFQxNzI0MTFaIl1dfQ== -F file=/home/ashish/Pictures/test.png https://play.min.io:9000/testbucket 
```
The XML received :
```
<?xml version="1.0" encoding="UTF-8"?>
<PostResponse><Bucket>testbucket</Bucket><Key>my-objectname</Key><ETag>&#34;d1237fb44ada3718026aeafc725e8402&#34;</ETag><Location>https://play.min.io:9000/testbucket/my-objectname</Location></PostResponse>
```

Closes #782

### CODE REVIEW

File: api/src/main/java/io/minio/PostPolicy.java [hunk 1]
```diff
@@ -91,6 +92,17 @@ public void setContentType(String contentType) throws InvalidArgumentException {
     this.contentType = contentType;
   }
 
+  /**
+   * Sets success action status.
+   */
+  public void setSuccessActionStatus(int successActionStatus) throws InvalidArgumentException {
+    if (!(successActionStatus == 200 || successActionStatus == 201 || successActionStatus == 204)) {
+      throw new InvalidArgumentException("Success action status accepts the values as 200, 201, or "
```
Comment (harshavardhana) [inline api/src/main/java/io/minio/PostPolicy.java]:
```suggestion
      throw new InvalidArgumentException("Invalid action status, acceptable values are 200, 201, or 204"
```

Comment (harshavardhana) [reply api/src/main/java/io/minio/PostPolicy.java]: Now to thinking more, we should have successActionStatus as enum value such as

```
successActionStatus enum {
   successActionStatus201
   successActionStatus200
   successActionStatus204
}
```

This way validation becomes easy.

Comment (sinhaashish) [reply api/src/main/java/io/minio/PostPolicy.java]: done.

**Comment by sinhaashish** [comment]
>How about we update the docs in API.md as well.
Updated it.

**Comment by poornas** [comment]
@sinhaashish , travis failures on this.

**Comment by sinhaashish** [comment]
> @sinhaashish , travis failures on this.

restarted the job and it passes.

**Comment by kannappanr** [comment]
ping @harshavardhana

**Pull request merged**